### PR TITLE
Dives deeper into mount integration

### PR DESCRIPTION
**Pull request by dposada**
If you haven't already, please see https://github.com/twosigma/Cook/pull/741 for context.

This PR builds on top of https://github.com/twosigma/Cook/pull/741 and shows what it looks like to remove the executor config argument from all of the call chains it's passed through. It also addresses the corresponding unit tests.

### CODE REVIEW

File: scheduler/src/cook/config.clj [hunk 1]
```diff
@@ -58,7 +58,6 @@
                              "com.netflix.fenzo.TaskScheduler" :warn
                              "com.netflix.fenzo.AssignableVirtualMachine" :warn}}))
   ([{:keys [file] :or {file "log/cook.log"} {:keys [default] :or {default :info} :as overrides} :levels}]
-   (println "Initializing logging")
```
Comment (shamsimam) [inline scheduler/src/cook/config.clj:61]: Shouldn't be part of this PR :)

File: scheduler/src/cook/config.clj [hunk 2]
```diff
@@ -355,7 +354,6 @@
   (try
     (let [literal-config {:config config-map}]
       (pre-configuration literal-config)
-      (.println System/err "Configured logging")
```
Comment (shamsimam) [inline scheduler/src/cook/config.clj:358]: Shouldn't be part of this PR :)

File: scheduler/src/cook/mesos/task.clj [hunk 3]
```diff
@@ -52,17 +54,19 @@
...
       Or: the job is a cook-executor candidate (see cook-executor-candidate?)."
-  [job-ent {:keys [command portion retry-limit]}]
-  (and (not (use-custom-executor? job-ent))
-       command
-       (or (= :executor/cook (:job/executor job-ent))
-           (cook-executor-candidate? job-ent portion retry-limit))))
+  [job-ent]
+  (let [{:keys [command]} (-> config :settings :executor)]
+    (and (not (use-custom-executor? job-ent))
+         command
```
Comment (shamsimam) [inline scheduler/src/cook/mesos/task.clj]: Can we use `(-> config :settings :executor :command)` to avoid the explicit `let` block.

**Review by shamsimam — APPROVED** [review]

File: scheduler/src/cook/config.clj (hunk 1 above)
Comment (dposada) [reply scheduler/src/cook/config.clj:61]: It's related to this PR because I found it very annoying to run the unit tests with these `println`s, and I don't think they're necessary anyway

File: scheduler/src/cook/config.clj (hunk 2 above)
Comment (dposada) [reply scheduler/src/cook/config.clj:358]: It's related to this PR because I found it very annoying to run the unit tests with these `println`s, and I don't think they're necessary anyway

File: scheduler/src/cook/mesos/task.clj (hunk 3 above)
Comment (dposada) [reply scheduler/src/cook/mesos/task.clj]: Good call, done
